models/pokemon.js

```diff
@@ -269,18 +271,104 @@ Pokemon.prototype.assignShiny = function () {
   this.shiny = Math.floor(Math.random() * SHINY_CHANCE + 1) >= SHINY_CHANCE;
 };
 
+// assign random moves based on level
+logger.info("[pokemon] Assigning random moves based on level");
+Pokemon.prototype.assignMoves = function () {
+
+    // video game level
+    let vgLevel = this.level * 5;
+
+    //moves the pokemon can legally learn
+    let legalMoves = [];
+    // make a blank move for use later if needed
+    let blankMove = new Moveset.Move();
+
+    // find all moves that can legally be learned by the pokemon
+    this.pokemonData.moves.forEach(nextMove =>
+    {
+        let newestAppearance = nextMove.version_group_details.length - 1;
+        let learnMethod = nextMove.version_group_details[newestAppearance].move_learn_method.name;
+        let levelLearned = nextMove.version_group_details[newestAppearance].level_learned_at;
+        if (learnMethod === "level-up" && levelLearned <= vgLevel) { legalMoves.push(nextMove) }
+    })
+
+    // if there aren't enough moves to fill out the known moves, grab the ones that do exist
+    if (legalMoves.length < 4) {
+        try {
+            console.log("moveloop");
+            this.moveSet.move1 = legalMoves[0];
+            this.moveSet.move2 = legalMoves[1];
+            this.moveSet.move3 = legalMoves[2];
+            this.moveSet.move4 = legalMoves[3];
+        } catch (e) {
+            if (!this.moveSet.move2) this.moveSet.move2 = blankMove;
+            if (!this.moveSet.move3) this.moveSet.move3 = blankMove;
+            if (!this.moveSet.move4) this.moveSet.move4 = blankMove;
+        }
+    } else{//roll four random numbers between 0 & # of moves found
+        let moveRoller = [];
+
+        //
+        while (moveRoller.length < 4) {
+            let randMove = Math.floor(Math.random() * legalMoves.length);
+            if (moveRoller.indexOf(randMove) === -1) moveRoller.push(randMove);
+        }
+
+        // for each random number picked, assign!
+        this.moveSet.move1 = legalMoves[moveRoller[0]];
+        this.moveSet.move2 = legalMoves[moveRoller[1]];
+        this.moveSet.move3 = legalMoves[moveRoller[2]];
+        this.moveSet.move4 = legalMoves[moveRoller[3]];
+    }
+}
+
 // capitalize words
 
 let capitalizeWord = function (tempWord) {
   return tempWord.charAt(0).toUpperCase() + tempWord.substr(1);
 };
 
+// format dashed stuff nicely
+let fixAbilityOrMoveFormatting = function (tempWord, middle) {
+    // set up formatted ability name
+    try {
+        if (~tempWord.indexOf("-")) {
+            // if the word is only a dash, return it
+            if (tempWord === "-") return tempWord;
+            let tempA = tempWord.slice(0, tempWord.indexOf("-"));
```

Comment:
There are def String functions that do what you're going for here- too tired to google them rn

Reply:
And something that can do it without having to repeat all this code

Reply:
But you know, when it works it works.

Reply:
Found it, + added new function that sets stuff to title case so there's 100% no need to break the string into chonks for formatting anymore.